### PR TITLE
envoy: Upstream rebase

### PR DESCRIPTION
**Pull request by jrajahalme**
Rebase to newer upstream Envoy to get tcp_proxy fixes.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5613)

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

File: envoy/BUILD_DEPS [hunk 1]
```diff
@@ -34,9 +34,5 @@ envoy_cc_binary(
         "@envoy//source/common/router:config_utility_lib",
         "@envoy//source/exe:envoy_main_entry_lib",
         "@envoy//source/exe:envoy_common_lib",
-        "@istio_proxy//src/envoy/http/authn:filter_lib",
```
Comment (rlenglet) [inline envoy/BUILD_DEPS:37]: Just out of curiosity, was it necessary to remove the Istio filters' deps? Now I think it's actually good to keep them, to make sure we don't break the compatibility with Istio.

**Review by rlenglet — APPROVED** [review]
LGTM. Just one question about keeping the Istio proxy filter deps.

File: envoy/BUILD_DEPS (hunk 1 above)
Comment (jrajahalme) [reply envoy/BUILD_DEPS:37]: I may have hastened the removal, as these broke the build when I had not yet updated the ISTIO_SHA. However, I'd introduce these back in a separate PR as it takes quite a while to have the packer-ci-image & cilium-builder images recreated, sorry.

Comment (rlenglet) [reply envoy/BUILD_DEPS:37]: No problem!

**Review by ianvernon — APPROVED** [review]

**Pull request merged**